src/metadata/align_query_reference/script.py

```diff
@@ -0,0 +1,230 @@
+    "reference_obs_batch": "donor_id",
+    "reference_obs_label": "cell_ontology_class",
+    "reference_id": "reference",
+    "output_query": "pbmc_1k_protein_v3_mms_ligned.h5mu",
+    "output_reference": "TS_Blood_filtered_aligned.h5mu",
+    "output_layer": "_counts",
+    "output_var_gene_names": "_gene_names",
+    "output_obs_batch": "_sample_id",
+    "output_obs_label": "_cell_type",
+    "output_obs_id": "_dataset",
+    "input_reference_gene_overlap": 100,
+    "unkown_celltype_label": "Unknown",
+    "overwrite_existing_key": False,
+    "output_compression": None,
+}
+
+meta = {"resources_dir": "src/utils"}
+## VIASH END
+
+sys.path.append(meta["resources_dir"])
+from setup_logger import setup_logger
+from cross_check_genes import cross_check_genes
+
+logger = setup_logger()
+
+
+def copy_layer(adata, input_layer, output_layer, overwrite=False):
+    if output_layer not in adata.layers.keys() and input_layer:
+        logger.info(f"Copying layer from `{input_layer}` to `{output_layer}`")
+        adata.layers[output_layer] = adata.layers[input_layer].copy()
+
+    elif output_layer not in adata.layers.keys() and not input_layer:
+        logger.info(f"Copying layer from .X to `{output_layer}`")
+        adata.layers[output_layer] = adata.X.copy()
+
+    else:
+        if not overwrite:
+            raise ValueError(
+                f"Layer `{output_layer}` already exists. Data can not be copied."
+            )
+        logger.warning(f"Layer `{output_layer}` already exists. Overwriting data.")
+        adata.layers[output_layer] = (
+            adata.layers[input_layer].copy() if input_layer else adata.X.copy()
+        )
+
+    return adata
+
+
+def copy_obs(adata, input_obs_key, output_obs_key, overwrite=False, fill_value=None):
+    if output_obs_key not in adata.obs.keys() and input_obs_key:
+        logger.info(f"Copying .obs field from `{input_obs_key}` to `{output_obs_key}`")
+        adata.obs[output_obs_key] = adata.obs[input_obs_key].copy()
+
+    elif output_obs_key not in adata.obs.keys() and not input_obs_key:
+        logger.info(f"Assigning fill value `{fill_value}` to .obs {output_obs_key}")
+        adata.obs[output_obs_key] = fill_value
+
+    else:
+        if not overwrite:
+            raise ValueError(
+                f".obs key `{output_obs_key}` already exists. Data can not be copied."
+            )
+
+        logger.warning(f".obs key `{output_obs_key}` already exists. Overwriting data.")
+        adata.obs[output_obs_key] = (
+            adata.obs[input_obs_key].copy() if input_obs_key else fill_value
+        )
+
+    return adata
+
+
+def copy_and_sanitize_var_gene_names(
+    adata, input_var_key, output_var_key, overwrite=False
+):
+    if output_var_key not in adata.var.keys() and input_var_key:
+        logger.info(f"Copying .var field from `{input_var_key}` to `{output_var_key}`")
+        adata.var[output_var_key] = [
+            re.sub("\\.[0-9]+$", "", s) for s in adata.var[input_var_key]
```

Comment:
I think its better to use the native pandas replace method here because its much more performant for large datasets: https://pandas.pydata.org/docs/reference/api/pandas.Series.str.replace.html 
Could you check if other instances of `re.sub` can also be replaced?